server/src/core/search/mod.rs

```diff
@@ -0,0 +1,58 @@
+mod search_executor;
```

Comment:
I dont like how the search directory is basically empty, but this is how rust does things..

Reply:
I don't have a lot of experience about this either, so maybe that's just fine.